pkg/gops/cell.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package gops
+
+import (
+	"context"
+	"fmt"
+
+	gopsAgent "github.com/google/gops/agent"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"go.uber.org/fx"
+
+	"github.com/cilium/cilium/pkg/defaults"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+// Gops runs the gops agent, a tool to list and diagnose Go processes.
+// See https://github.com/google/gops.
+var Cell = hive.NewCellWithConfig[GopsConfig](
+	"gops",
+	fx.Invoke(registerGopsHooks),
+)
+
+type GopsConfig struct {
+	GopsPort uint16 // Port for gops server to listen on
+}
+
+func (GopsConfig) CellFlags(flags *pflag.FlagSet) {
+	flags.Uint16(option.GopsPort, defaults.GopsPortAgent, "Port for gops server to listen on")
```

Comment:
For this to be reused by other packages, we'll need to make this default gops port configurable :thinking:

Reply:
Ah yep, good point. Not quite sure what'd be a good pattern to use here. A simple option is having the cell define its defaults via globals that can be mutated from the outside, but then you'd need to think about when `CellFlags` gets invoked. Can also be worked around after `hive.New` by modifying the `Flag.DefValue` and `Flag.Value`, but that gets ugly quickly. Fairly easy to solve this once we hit an issue with reuse.